File: Codigos/Ejercicio5.java
import java.util.ArrayList;
import java.util.Scanner;
 
public class Ejercicio5 {
    public static void main(String[] args) {
        ArrayList<String> nombres = new ArrayList<>();
        ArrayList<Double> notas   = new ArrayList<>();
        Scanner sc = new Scanner(System.in);
 
        System.out.print("¿Cuantos estudiantes? ");
        int n = sc.nextInt();
 
        for (int i = 0; i < n; i++) {
            System.out.print("Nombre: ");    nombres.add(sc.next());
            System.out.print("Nota (0-10): "); notas.add(sc.nextDouble());
        }
 
        System.out.println("\n--- Listado ---");
        for (int i = 0; i < nombres.size(); i++) {
            String estado = notas.get(i) >= 7 ? "APROBADO" : "REPROBADO";
            System.out.println(nombres.get(i) + " - " + notas.get(i) + " [" + estado + "]");
        }
        sc.close();
    }
}

File: Codigos/Ejercicio5.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
 
struct Estudiante { string nombre; float nota; };
 
int main() {
    vector<Estudiante> lista;
    int n;
    cout << " Nombres y Notas con Estado "<< endl;
    cout << "¿Cuantos estudiantes? "; cin >> n;
 
    for (int i = 0; i < n; i++) {
        Estudiante e;
        cout << "Nombre: "; cin >> e.nombre;
        cout << "Nota (0-10): "; cin >> e.nota;
        lista.push_back(e);
    }
 
    cout << "\n--- Listado ---" << endl;
    for (auto& e : lista)
        cout << e.nombre << " - Nota: " << e.nota
             << (e.nota >= 7 ? " [APROBADO]" : " [REPROBADO]") << endl;
    return 0;
}
